girder/utility/s3_assetstore_adapter.py

```diff
@@ -727,45 +575,33 @@ def makeBotoConnectParams(accessKeyId, secret, service=None):
 
     :param accessKeyId: the S3 access key ID
     :param secret: the S3 secret key
-    :param service: the name of the service in the form
-                    [http[s]://](host domain)[:(port)].
+    :param service: alternate service URL
     :returns: boto connection parameter dictionary.
     """
     if accessKeyId and secret:
-        connect = {
+        params = {
             'aws_access_key_id': accessKeyId,
             'aws_secret_access_key': secret,
-            }
+            'config': botocore.client.Config(signature_version='s3v4')
+        }
     else:
-        connect = {
-            'anon': True
+        params = {
+            'config': botocore.client.Config(signature_version=botocore.UNSIGNED)
```

Comment:
This instructs boto that it should not attempt to authenticate its requests. This was extremely tricky in boto 2, but this is all it takes for it to work in boto3. It's possible that this simply fixes #2054, but I haven't verified that.